### ISSUE TITLE
webcrypto-shim.js: bad algorithm name with FS2.4 and IE11

### ISSUE THREAD

**Issue by fransward** [filesender/filesender#452]
We discovered another problem after we updated to Filesender 2.4. Several users are complaining that they cannot use file encryption when using IE11.

Just tested myself and the upload process fails after a while with seemingly no progress:
"No progress has been reported for your currently uploading transfer for some time, if you think it is tuck and before reporting it to your support team please send the last log entries from your user interface by clicking this button: [send logs]"

The IE debugger window shows:
`throw new SyntaxError(Bad algorithm name");
Exception thrown and not caught`

**Comment by fransward** [comment]
Also when you receive an encrypted file IE11 does not decrypt when the sender used another browser.

**Comment by monkeyiq** [comment]
The IE11 incompatibility was introduced to address a security issue raised at https://github.com/filesender/filesender/issues/338 which recommends using deriveKey().

Unfortunately deriveKey() is not supported in IE11 and the shim code to call the web crypto api does not give a very meaningful error. Although the documentation (https://github.com/filesender/filesender/blob/master/www/vendor/node_modules/webcrypto-shim/README.md) mentions that ie11 and safari lack deriveKey support I found using the test page (https://vibornoff.github.io/webcrypto-examples/index.html) that safari supports PBKDF2 + deriveKey. I have also just performed another test upload on encrypted content using safari on a recent osx machine to confirm. On the other hand, on current Windows 10 using its IE11 version there is no deriveKey in any row on the compatibility page.

I will put an issue on github to allow the crypto version to be set in config.php instead of in the code itself for new files so that admins can choose which key code to use in order to support the browsers they wish to support.
